include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -378,15 +365,13 @@ ORT_API(size_t, OrtGetNumOfDimensions, _In_ const OrtTensorTypeAndShapeInfo* inf
 ORT_API(void, OrtGetDimensions, _In_ const OrtTensorTypeAndShapeInfo* info, _Out_ int64_t* dim_values, size_t dim_values_length);
 
 /**
- * How many elements does this tensor have.
- * May return a negative value
+ * Return the number of elements specified by the tensor shape.
+ * Return a negative value if unknown (i.e., any dimension is negative.)
```

Comment:
In the onnx spec, no dimension could be negative. Negative values came from symbolic dims. We treat symbolic dim as -1 internally, and looks like it's not quite correct.